Library/Homebrew/formulary.rb

```diff
@@ -108,7 +120,97 @@ def self.load_formula_from_path(name, path, flags:, ignore_errors:)
     contents = path.open("r") { |f| ensure_utf8_encoding(f).read }
     namespace = "FormulaNamespace#{Digest::MD5.hexdigest(path.to_s)}"
     klass = load_formula(name, path, contents, namespace, flags: flags, ignore_errors: ignore_errors)
-    cache[path] = klass
+    cache[:path] ||= {}
+    cache[:path][path] = klass
+  end
+
+  def self.load_formula_from_api(name, flags:)
+    namespace = "FormulaNamespaceAPI#{Digest::MD5.hexdigest(name)}"
+
+    mod = Module.new
+    remove_const(namespace) if const_defined?(namespace)
+    const_set(namespace, mod)
+
+    mod.const_set(:BUILD_FLAGS, flags)
+
+    class_s = Formulary.class_s(name)
+    json_formula = Homebrew::API::Formula.all_formulae[name]
+
+    klass = Class.new(::Formula) do
+      desc json_formula["desc"]
+      homepage json_formula["homepage"]
+      license json_formula["license"]
+      revision json_formula["revision"]
+      version_scheme json_formula["version_scheme"]
+
+      if (urls_stable = json_formula["urls"]["stable"]).present?
+        stable do
+          url urls_stable["url"]
+          version json_formula["versions"]["stable"]
+        end
+      end
+
+      if (bottles_stable = json_formula["bottle"]["stable"]).present?
+        bottle do
+          root_url bottles_stable["root_url"]
+          rebuild bottles_stable["rebuild"]
+          bottles_stable["files"].each do |tag, bottle_spec|
+            cellar = bottle_spec["cellar"]
+            cellar = cellar[1..].to_sym if cellar.start_with?(":")
+            sha256 cellar: cellar, tag.to_sym => bottle_spec["sha256"]
+          end
+        end
+      end
+
+      if (keg_only_reason = json_formula["keg_only_reason"]).present?
+        reason = keg_only_reason["reason"]
+        reason = reason[1..].to_sym if reason.start_with?(":")
+        keg_only reason, keg_only_reason["explanation"]
```

Comment:
Would be good (in future PR) to ensure all these lines are covered by tests. Should be fairly easy to give it a comprehensive enough Ruby Hash as input.

Reply:
Honestly, I can probably just add those now. I already added a test just not one that tests these particular parts since they're optional.

Reply:
Thanks!